### ISSUE TITLE
academic consult for code 

### ISSUE THREAD

**Issue by labiao** [WangLukang/CS-WSCDNet#1]
I appreciate the quality of your work and have reviewed the code in SAM_label.py. I have a couple of questions: Firstly, should lines 79-81 be included within a for loop? Additionally, I noticed that in line 84, when calculating overlaps, the 'mask' variable gets overwritten and does not seem to represent BM1 accurately.

**Comment by WangLukang** [comment]
You're absolutely right, and I apologize for these oversights in the code. Thank you for your careful review and feedback. I'll promptly revise the code to address these issues.
